### PR TITLE
Add hadolint checker

### PR DESCRIPTION
**Pull request by fauust**
Fix some warnings/errors

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -22,15 +22,16 @@ RUN set -eux; \
 	apt-get install -y --no-install-recommends wget; \
 	rm -rf /var/lib/apt/lists/*; \
 	dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
-	wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
-	wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
-	export GNUPGHOME="$(mktemp -d)"; \
+	wget -q -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
+	wget -q -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
```
Comment (fauust) [inline Dockerfile.template:26]: Not sure about this one: https://github.com/hadolint/hadolint/wiki/DL3047
So DL3047 could be disabled.

Comment (grooverdan) [reply Dockerfile.template:26]: this is fine.

**Comment by grooverdan** [comment]
There should be a note about the impact of `--no-install-recommends` in the commit message.

Targetting `next` branch so that it gets buildbot testing before release.

**Comment by fauust** [comment]
> There should be a note about the impact of `--no-install-recommends` in the commit message.
> 
> Targetting `next` branch so that it gets buildbot testing before release.

Yep, indeed, good catch. I was not sure about it and forget to mention. We can also remove it and skip this check.

**Comment by grooverdan** [comment]
Thank you.

**Pull request merged**